pkg/server/plugin/upstreamauthority/spire/spire_test.go

```diff
@@ -307,12 +319,15 @@ func TestSpirePlugin_PublishJWTKey(t *testing.T) {
 	m, done := newWithDefault(t, server.napiServer.addr, server.wapiServer.socketPath)
 	defer done()
 
-	resp, err := m.PublishJWTKey(context.Background(), &upstreamauthority.PublishJWTKeyRequest{
+	stream, err := m.PublishJWTKey(context.Background(), &upstreamauthority.PublishJWTKeyRequest{
 		JwtKey: &common.PublicKey{
 			Kid: "kid-1",
 		},
 	})
+	require.NoError(t, err)
+	require.NotNil(t, stream)
 
+	resp, err := stream.Recv()
```

Comment:
We should add an additional call to Recv() somewhere after here that we assert returns io.EOF.